Replicator/Pusher.cc

```diff
@@ -627,14 +627,14 @@ namespace litecore { namespace repl {
         // immediate means I want to resend as soon as possible, bypassing another changes feed entry
         // (for example in the case of a failed delta merge)
         logInfo("%d documents failed to push and will be retried now", int(revsToRetry.size()));
-        _caughtUp = false;
         if (immediate) {
             for (const auto& revToRetry : revsToRetry) {
                 _pushingDocs.insert({revToRetry->docID, revToRetry});
                 addProgress({0, revToRetry->bodySize});
             }
             _revQueue.insert(_revQueue.begin(), revsToRetry.begin(), revsToRetry.end());
         } else {
+            _caughtUp = false;
```

Comment:
Can you describe about this fix as well?

Reply:
In immediate case, we take them directly to _revQueue, not going through the changes feed, and _caughtUp describes whether the changes feed has all ben taken out. That is, immediate case does not affect _caughtUp.

Reply:
This is the cause of hanging Push. It does not contributing to the CBSE case, which is Pull only.

Reply:
Got it. Thanks!

Reply:
If there is not a CBL ticket for this change, please create one before this is committed

Reply:
This fix only affects Push side. The original CBSE is related to Pull. That is, it won't affect CBSE per se. However, without the fix, our localReplicator test will fail, because in this case, CBL plays both sides of Push and Pull.